mitodl/mit-open
# Show "Start Anytime" based on resource "availability" property

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4743

### Description (What does it do?)
This uses the new "availability" property on resources to display "Starts: Anytime" on resource cards.

### Screenshots (if appropriate):

**Vertical Cards, small**
<img width="433" alt="Screenshot 2024-07-30 at 9 47 13 AM" src="https://github.com/user-attachments/assets/d6db9ae8-70a9-417e-9e0a-0a1a77d4542e">

**Vertical Cards, medium**
<img width="647" alt="Screenshot 2024-07-30 at 10 11 50 AM" src="https://github.com/user-attachments/assets/b0b5480f-95ca-41c0-ad60-673724da64b8">

**Horizontal, small screen**
<img width="367" alt="Screenshot 2024-07-30 at 11 05 13 AM" src="https://github.com/user-attachments/assets/3e217dca-5ff4-4f4c-a971-e015eddc0c20">

**Horizontal, wide screen**
<img width="1000" alt="Screenshot 2024-07-30 at 11 04 34 AM" src="https://github.com/user-attachments/assets/68647ddc-295e-4058-a631-612d18f11386">

**Drawer:**
<img width="350" alt="Screenshot 2024-07-30 at 9 55 34 AM" src="https://github.com/user-attachments/assets/94a477f6-dc8d-4f3d-ba19-9ee521be98fa"> <img width="350" alt="Screenshot 2024-07-30 at 9 56 04 AM" src="https://github.com/user-attachments/assets/e3386a6b-da89-44a7-b671-0858e93e4ac1">



### How can this be tested?
1. If you haven't re-run backpopulates since #1301 , run them using RC creds.
    ```sh
    ./manage.py backpopulate_ocw_data --skip-contentfiles
    ./manage.py backpopulate_mitxonline_data
    ./manage.py backpopulate_xpro_data
    ./manage.py backpopulate_prolearn_data
    ./manage.py backpopulate_oll_data
    # The edX API has a rate limit; this may 429 if other people (or RC) has run it recently.
    ./manage.py extract_openedx_data
    ```
2. View course cards and resource drawer. If the API returns `available: "anytime"` for courses, you should see "Starts: Anytime". Otherwise, you'll see a date like before. Notes:
    - We previously showed "As taught in..." on cards for OCW courses. We do not anymore.
    - "Starts Anytime" is NOT limited to OCW (but is limited to courses and programs in the UI).
    - Card locations:
        - vertical cards:  homepage, unit channel pages, and dashboard for **vertically oriented cards** (Dashboard has small cards)
        - horizontal cards: channel page, search page
3. Check the resource drawer (click any card). If a card displays "Start anytime", it should display "As taught in..." in the resource drawer.
4. View new storybook stories: http://localhost:6006/?path=/story/smoot-design-cards-learningresourcecard--start-anytime (and dated nearby), and similar stories for the list card.
